File: typescript/6kyu/Find_The_Parity_Outlier.ts
export const findOutlier = (integers: number[]): number => {
  const even = integers.filter((a) => a % 2 === 0);
  const odd = integers.filter((a) => a % 2 !== 0);
  return even.length === 1 ? even[0] : odd[0];
};

File: rust/src/6kyu/Find_The_Parity_Outlier.rs
fn find_outlier(s: &[i32]) -> i32 {
    let is_even = |x: &&_| *x % 2 == 0;
    let is_odd = |x: &&_| *x % 2 != 0;
    *s.iter()
        .find(if s.iter().take(3).filter(is_even).count() > 1 {
            is_odd
        } else {
            is_even
        })
        .unwrap()
}
